### PR TITLE
[realm-core] new port

### PR DESCRIPTION
**Pull request by leemaguire**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -979,6 +979,7 @@ rbdl-orb:x64-osx=skip
 rbdl:arm-neon-android=fail
 rbdl:arm64-android=fail
 rbdl:x64-android=fail
+realm-core:arm64-uwp=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:982]: @leemaguire, thanks for the PR, could you clarify that triples are not supported upstream?

**Comment by JonLiu1993** [comment]
Ping @leemaguire for response.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (leemaguire) [reply scripts/ci.baseline.txt]: I think I can resolve the issue with a patch and not actually exclude this platform.

**Comment by leemaguire** [comment]
Hi @JonLiu1993 I was on vacation the past two weeks, I'm getting back to this now.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by realm-core:x64-windows:
````
realm-core provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(Realm CONFIG REQUIRED)
  target_link_libraries(main PRIVATE Realm::Sync Realm::Storage Realm::ObjectStore Realm::QueryParser)
````

**Review by data-queue — APPROVED** [review]

**Pull request merged**